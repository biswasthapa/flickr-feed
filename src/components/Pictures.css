.spinner-loading {
  position: absolute;
  margin: 0 auto;
  top: 20%;
  left: 45%;
}

.text-tag {
  color: #456dc4;
}
