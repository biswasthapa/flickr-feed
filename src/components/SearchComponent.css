.search-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100%;
  width: 100%;
  margin: 0 auto;
  min-height: 100%;
  min-height: 100vh;
}

.search-container h1 {
  color: #0063dc;
  font-size: 24px;
  margin: 0 0 10px;
}

.search-container h1 b {
  color: #fe0083;
}

.search-container .search-text {
  width: 100%;
  margin: 0 0 10px;
}

.search-container .btn {
  margin-top: 10px;
}

.search-container.search-top .btn {
  margin: 10px 0px 20px 10px;
}

.search-container.search-top {
  height: auto;
  width: 100%;
  flex-direction: row;
  min-height: 0;
}

.search-container.search-top h1 {
  font-size: 10px;
}

.search-block.search-top .input {
  border-radius: 0px;
  padding: 11px 30px;
}

.form-container {
  width: 100%;
  height: 100%;
}
