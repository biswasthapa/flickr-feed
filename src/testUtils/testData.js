export const flickrTestData = {
  title: "Recent Uploads tagged ddd",
  link: "https://www.flickr.com/photos/tags/ddd/",
  description: "",
  modified: "2020-01-23T22:00:29Z",
  generator: "https://www.flickr.com",
  items: [
    {
      title: "TDD Thursday Door Day / DDD Donderdag Deuren Dag",
      link: "https://www.flickr.com/photos/jomisere/49427089937/",
      media: {
        m: "https://live.staticflickr.com/65535/49427089937_7f61907ed3_m.jpg"
      },
      date_taken: "2015-06-26T14:38:30-08:00",
      description:
        ' <p><a href="https://www.flickr.com/people/jomisere/">jo.misere</a> posted a photo:</p> <p><a href="https://www.flickr.com/photos/jomisere/49427089937/" title="TDD Thursday Door Day / DDD Donderdag Deuren Dag"><img src="https://live.staticflickr.com/65535/49427089937_7f61907ed3_m.jpg" width="160" height="240" alt="TDD Thursday Door Day / DDD Donderdag Deuren Dag" /></a></p> <p>Join the Group / one door a week / Donderdag /Thursday<br /> Het hoofdbestanddeel moet een hele deur zijn /geen personen<br /> The main component has to be a door / No persons</p>',
      published: "2020-01-23T22:00:29Z",
      author: 'nobody@flickr.com ("jo.misere")',
      author_id: "32698450@N06",
      tags: "ddd tdd dtt doors deuren"
    },
    {
      title: "One line of light through the door...TDD/DDD",
      link: "https://www.flickr.com/photos/105859735@N04/49429597302/",
      media: {
        m: "https://live.staticflickr.com/65535/49429597302_08c6c0d71a_m.jpg"
      },
      date_taken: "2020-01-18T09:30:35-08:00",
      description:
        ' <p><a href="https://www.flickr.com/people/105859735@N04/">wilma HW61</a> posted a photo:</p> <p><a href="https://www.flickr.com/photos/105859735@N04/49429597302/" title="One line of light through the door...TDD/DDD"><img src="https://live.staticflickr.com/65535/49429597302_08c6c0d71a_m.jpg" width="240" height="179" alt="One line of light through the door...TDD/DDD" /></a></p> <p>THANKS FOR YOUR VISIT AND FAVES<br /> ON THE REACTIONS I WILL TRY TO RESPOND BACK</p>',
      published: "2020-01-23T13:20:49Z",
      author: 'nobody@flickr.com ("wilma HW61")',
      author_id: "105859735@N04",
      tags:
        "tdd ddd deur door t\u00fcr porte porta light licht lichtval streep stripe fabriek factory factoryhall philips zwolle overijssel photowalkoverijssel urban ub urbex nederland niederlande netherlands nikond90 holland holanda paysbas paesibassi pa\u00edsesbajos europa europe indoor dwwg"
    },
    {
      title: "Hospice Wallon",
      link: "https://www.flickr.com/photos/komeet/49428336423/",
      media: {
        m: "https://live.staticflickr.com/65535/49428336423_b292059714_m.jpg"
      },
      date_taken: "2020-01-16T14:54:09-08:00",
      description:
        ' <p><a href="https://www.flickr.com/people/komeet/">Hans Veuger</a> posted a photo:</p> <p><a href="https://www.flickr.com/photos/komeet/49428336423/" title="Hospice Wallon"><img src="https://live.staticflickr.com/65535/49428336423_b292059714_m.jpg" width="180" height="240" alt="Hospice Wallon" /></a></p> <p>Amsterdam - Vijzelgracht.<br /> <br /> The Hospice Wallon (Walenweeshuis) was built in 1683.<br /> The Institut fran\u00e7ais des Pays-Bas moved to this premises in 1971.<br /> <br /> Het Institut fran\u00e7ais des Pays-Bas (voorheen Maison Descartes) was het Franse Instituut voor taal en cultuur in Nederland, vanaf 1971 gevestigd in het oude Walenweeshuis uit 1683.<br /> Parijs is in 2016 gestopt met de verspreiding van de Franse cultuur vanuit Amsterdam. Het pand is verkocht.<br /> <br /> TDD / DDD</p>',
      published: "2020-01-23T10:12:59Z",
      author: 'nobody@flickr.com ("Hans Veuger")',
      author_id: "64472650@N05",
      tags:
        "nederland thenetherlands amsterdam amsterdamcentrum vijzelgracht ddd tdd door hospicewallon walenweeshuis maisondescartes deur nikon b700 coolpix nederlandvandaag unlimitedphotos twop 1683"
    }
  ]
};
