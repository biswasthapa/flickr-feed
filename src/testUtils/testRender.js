import React from "react";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { mount } from "enzyme";

export default (defaultState = {}) => {
  const mockStore = configureMockStore([thunk]);

  return {
    mountWithProvider: (component, initialState = {}) => {
      const store = mockStore(Object.assign({}, defaultState, initialState));
      return mount(<Provider store={store}>{component}</Provider>);
    }
  };
};
