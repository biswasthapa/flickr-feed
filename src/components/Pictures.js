import React from "react";
import strftime from "strftime";
import { Icon } from "@blueprintjs/core";
import { IconNames } from "@blueprintjs/icons";
import "./Pictures.css";

export const Pictures = ({ photos }) => (
  <div className="row">
    <div className="card-columns photos-list">
      {photos.map(photo => (
        <div key={photo.media.m} className="card">
          <div className="card-header bg-light">
            <small>
              <Icon icon={IconNames.USER} />
              &nbsp;
              {photo.author}
            </small>
          </div>
          <a href={`${photo.link}`} target="_blank" rel="noopener noreferrer">
            <img
              className="card-img-top img-fluid"
              src={photo.media.m}
              alt={photo.title}
            />
          </a>
          <div className="card-body">
            <p className="card-text">
              <small className="text-tag">
                #{photo.tags.split(" ").join(" #")}
              </small>
            </p>
          </div>
          <div className="card-footer">
            <small className="text-muted">
              Added on {strftime("%Y-%m-%d", new Date(photo.date_taken))}
            </small>
          </div>
        </div>
      ))}
    </div>
  </div>
);

export default Pictures;
