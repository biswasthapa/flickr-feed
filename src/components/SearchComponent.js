import React from "react";
import cx from "classnames";

import "./SearchComponent.css";

export const SearchComponent = ({ displayTop, onChange, onClick }) => {
  return (
    <div className={cx("search-container", { "search-top": displayTop })}>
      <h1>
        Search <b>Flickr!</b>
      </h1>
      <input
        type="text"
        className="form-control search-text"
        onChange={onChange}
        placeholder="Search using image tags..."
      />
      <span className="input-group-btn">
        <button
          className="btn btn-primary search-btn"
          type="submit"
          onClick={onClick}
        >
          Go!
        </button>
      </span>
    </div>
  );
};

export default SearchComponent;
