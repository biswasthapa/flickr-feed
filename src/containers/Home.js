import React, { Component } from "react";
import { connect } from "react-redux";
import { get } from "lodash";
import ClipLoader from "react-spinners/ClipLoader";
import { fetchPictures } from "../actions/photosActions";
import { Pictures } from "../components/Pictures";
import { SearchComponent } from "../components/SearchComponent";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tags: ""
    };
  }
  async searchFlickr(e, tags) {
    e.preventDefault();
    if (tags !== "") {
      await this.props.fetchPictures(tags);
    }
  }
  updateSearchText(newValue) {
    this.setState({ tags: newValue });
  }
  render() {
    return (
      <div className="container">
        <form>
          <SearchComponent
            displayTop={this.props.tags.length}
            onChange={e => this.updateSearchText(e.target.value)}
            onClick={e => this.searchFlickr(e, this.state.tags)}
          />
        </form>
        {this.props.photos.length > 0 && (
          <Pictures photos={this.props.photos} />
        )}
        {this.props.loading && (
          <div className="spinner-loading">
            <ClipLoader
              size={100}
              color={"#0063dc"}
              loading={this.props.loading}
            />
          </div>
        )}
      </div>
    );
  }
}

const mapDispatchToProps = { fetchPictures };

const mapStateToProps = store => {
  return {
    tags: get(store, "flickrPhotos.tags", ""),
    photos: get(store, "flickrPhotos.photos", []),
    loading: get(store, "flickrPhotos.loading", false)
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
