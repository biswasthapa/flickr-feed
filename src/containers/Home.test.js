import React from "react";
import ClipLoader from "react-spinners/ClipLoader";
import Home from "./Home";
import { Pictures } from "../components/Pictures";
import TestProviderSupport from "../testUtils/testRender";
import { flickrTestData } from "../testUtils/testData";
const { mountWithProvider } = TestProviderSupport();

describe("Home Container", () => {
  const state = {
    flickrPhotos: { tags: "", photos: [], loading: false }
  };

  it("should render", () => {
    const wrapper = mountWithProvider(<Home />, state);
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render Pictures component if photos list is empty", () => {
    const wrapper = mountWithProvider(<Home />, state);
    expect(wrapper.find(Pictures)).not.toExist();
  });

  it("should render Pictures component if photos list is not empty", () => {
    const updatedState = {
      ...state,
      flickrPhotos: {
        ...state.flickrPhotos,
        photos: flickrTestData.items
      }
    };
    const wrapper = mountWithProvider(<Home />, updatedState);
    expect(wrapper.find(Pictures)).toExist();
  });

  it("should not render the spinner if loading is false", () => {
    const wrapper = mountWithProvider(<Home />, state);
    expect(wrapper.find(ClipLoader)).not.toExist();
  });

  it("should render the spinner if loading is true", () => {
    const updatedState = {
      ...state,
      flickrPhotos: {
        ...state.flickrPhotos,
        loading: true
      }
    };
    const wrapper = mountWithProvider(<Home />, updatedState);
    expect(wrapper.find(ClipLoader)).toExist();
  });
});
