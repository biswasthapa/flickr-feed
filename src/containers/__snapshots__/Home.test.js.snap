// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container should render 1`] = `
<Connect(Home)>
  <div
    className="container"
  >
    <form>
      <div
        className="search-container"
      >
        <h1>
          Search 
          <b>
            Flickr!
          </b>
        </h1>
        <input
          className="form-control search-text"
          onChange={[Function]}
          placeholder="Search using image tags..."
          type="text"
        />
        <span
          className="input-group-btn"
        >
          <button
            className="btn btn-primary search-btn"
            onClick={[Function]}
            type="submit"
          >
            Go!
          </button>
        </span>
      </div>
    </form>
  </div>
</Connect(Home)>
`;
